client/src/components/HomePage/Registration.jsx

```diff
@@ -42,13 +42,21 @@ class Registration extends Component {
       .then(data => {
         this.setState({ msg: data.msg });
         window.Materialize.toast(this.state.msg, 3000);
-
+        console.log(data)
         if ('error' in data) {
           this.setState({ [`check_${data.error}`]: `Invalid ${data.error}` })
-          this.props.history.push('/registration');
           return null;
         } else {
-          this.props.history.push('/');
+          this.setState({
```

Comment:
I tested that on my branch. If you double clicks on button registration your page will be crush. Delete "this.setState" "here.

Reply:
Ok